.github/workflows/roll_torch_mlir.yml

```diff
@@ -0,0 +1,60 @@
+name: Roll PyTorch
+
+on:
+  schedule:
+    - cron: '0 13 * * *'
+  workflow_dispatch:
+
+jobs:
+  build_linux:
+    name: roll torch-mlir
+    needs: setup
+    runs-on:
+      - self-hosted
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+
+    # Don't run this in everyone's forks.
+    if: github.repository == 'open-xla/iree'
+
+    steps:
+
+      - name: Prepare workspace
+        run: |
+          # Clear the workspace directory so that we don't run into errors about
+          # existing lock files.
+          sudo rm -rf $GITHUB_WORKSPACE/*
+
+      - name: Get iree
+        uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          token: ${{ secrets.WRITE_ACCESS_TOKEN }}
+          ref: main
+          fetch-depth: 0
+
+      - name: update torch_mlir submodule
+        run: |
+          git config --local user.email "iree-github-actions-bot@google.com"
+          git config --local user.name "Roll torch-mlir"
+          git switch -c "rolltorchmlir_${{ github.run_number }}"
+          git submodule update --init --recursive third_party/torch-mlir
+          cd third_party/torch-mlir
+          git fetch --tags origin
+          git checkout origin/main
+          cd ../..
+          git add third_party/torch-mlir
+          git commit -m "update 'third_party/torch-mlir'"
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v5.0.1
```

Comment:
Same as the reason above (https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions), it's recommended to pin at a SHA